common/configdir.c

```diff
@@ -241,6 +244,33 @@ void parse_config_files(const char *config_filename,
 	parse_implied_config_file(config_basedir, chainparams->network_name, early);
 }
 
+/* Could be a yet-to-be-upgraded dir (definitely testnet), or could be
+ * it's been upgraded to testnet. */
+static bool smells_like_old_testnet(const char *config_basedir)
+{
+	struct stat st;
+	const char *base = default_base_configdir(NULL);
+
+	if (!config_basedir)
+		config_basedir = base;
+
+	/* If it doesn't exist, it's not testnet. */
+	if (stat(base, &st) != 0) {
+		tal_free(base);
+		return false;
+	}
+
+	/* Does it have a bitcoin/ subdir and no testnet/ subdir? */
+	if (stat(path_join(base, config_basedir, "bitcoin"), &st) == 0
```

Comment:
ok i'll admit i'm a bit confused here. at L255 we might have set `config_basedir` to be `base`; in that case we'd be joining the same path twice.

Reply:
First arg is context: we're not relying on tmpctx here, so I'm abusing base as a per-function ctx.  Will comment :)